### PR TITLE
Updated the annotations in QueryReportConfig fixing the query manager issue due to empty query language

### PR DESCRIPTION
**Pull request by ankit-gubrani**
Updated the annotations in QueryReportConfig fixing the query manager issue due to empty query language : Issue #3295

### CODE REVIEW

**Comment by masmohammed** [comment]
+1

**Review by masmohammed — APPROVED** [review]

File: bundle/src/main/java/com/adobe/acs/commons/reports/models/package-info.java [hunk 1]
```diff
@@ -15,7 +15,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-@org.osgi.annotation.versioning.Version("1.2.0")
+@org.osgi.annotation.versioning.Version("1.2.1")
```
Comment (davidjgonzalez) [inline bundle/src/main/java/com/adobe/acs/commons/reports/models/package-info.java:18]:
```suggestion
@org.osgi.annotation.versioning.Version("1.2.0")
```

Why does this require a version bump to 1.21? Im not seeing where the public API is changed in this package. Am i missing something?

**Review by davidjgonzalez — CHANGES_REQUESTED** [review]
Looks good other than the package-info bump... just want to confirm that bnd is asking for this -- im not seeing why it would.

File: bundle/src/main/java/com/adobe/acs/commons/reports/models/package-info.java (hunk 1 above)
Comment (kwin) [reply bundle/src/main/java/com/adobe/acs/commons/reports/models/package-info.java:18]: Micro version adjustments are required for almost all changes. You should never modify bytecode in a package without adjusting at least micro version.

**Comment by kwin** [comment]
I don’t understand what this fixes, by just switching to injector specific annotations? Which value was not correctly injected with `@Inject`?

**Comment by davidjgonzalez** [comment]
IIRC Sling Bindings is the only injector that precedes ValueMap .. so assuming this fixes it, `queryLanguage` (https://github.com/Adobe-Consulting-Services/acs-aem-commons/pull/3296/files#diff-59c585809b6d7e2373fbf6a355cd03425c103bd3aebd478b5cfa16820238d3bbR36) would be set in sling bindings?

I do like the change - as i always find @Inject unclear wrt to where the value is coming from -- but agreed, it's odd this is the fix.

@ankit-gubrani you did very this resolves the issue, correct?

**Comment by kwin** [comment]
@davidjgonzalez You can check via `/system/console/scriptingvariables` if a scripting variable with name `queryLanguage` does exist for a particular resource. I would be interested which Scripting Variables Provider would add such a variable (and with which value that is added which is not supported by the QueryManager....)

**Comment by ankit-gubrani** [comment]
> IIRC Sling Bindings is the only injector that precedes ValueMap .. so assuming this fixes it, `queryLanguage` (https://github.com/Adobe-Consulting-Services/acs-aem-commons/pull/3296/files#diff-59c585809b6d7e2373fbf6a355cd03425c103bd3aebd478b5cfa16820238d3bbR36) would be set in sling bindings?
> 
> I do like the change - as i always find @Inject unclear wrt to where the value is coming from -- but agreed, it's odd this is the fix.
> 
> @ankit-gubrani you did very this resolves the issue, correct?

Yes @davidjgonzalez , I did verify it fixes the issue. I initially updated the annotations in QueryReportConfig.java but then reports were showing data on executing the report but download was empty. So in order to fix that I had to update the annotations all the other places.

**Comment by ankit-gubrani** [comment]
> @davidjgonzalez You can check via `/system/console/scriptingvariables` if a scripting variable with name `queryLanguage` does exist for a particular resource. I would be interested which Scripting Variables Provider would add such a variable (and with which value that is added which is not supported by the QueryManager....). In a standard AEM cloud SDK server no such variable exists.

@kwin I didn't find a script variable named 'queryLanguage'. 
I have noticed with last few releases/versions of AEMaaCS @Inject annotation specially is not able to read resource properties, either we had to add @Named or diectly use @ValueMapValue, although I am not sure if others are also facing the similar issue but we ran into very similar issue few weeks back with a vendor provided translation connector as well.

**Comment by kwin** [comment]
If Inject annotations do no longer work, then this is a bug in AEMaaCS. Please raise a ticket with Adobe support to get that fixed.

**Comment by ankit-gubrani** [comment]
> I don’t understand what this fixes, by just switching to injector specific annotations? Which value was not correctly injected with `@Inject`?

So @Inject is not able to read the properties from JCR nodes in the Sling Models and thats why I initially thought it was just the queryLanguage which was empty but everything was empty so I had to update all the sling models later in my 2nd commit.

**Pull request merged**